### PR TITLE
post endpoint pt2

### LINKED ISSUE #75: Add mapping of error responses to POST endpoint
In general add more metadata that fast api supports

Tracked in Jira:OSS-305

### LINKED ISSUE #77: Switch email comparison to `.compare()` instead of `__eq__`
As discussed in the comment that will be linked below

Tracked in Jira:OSS-305

### PR DESCRIPTION
**Pull request by imbstack**
## Proposed changes

- Add 409 to responses for post. Fixes #75
- Make all schemas comparable. Fixes #77
- Return redirect from POST
- Check writes with more samples
- Improve schema model comparisons
- Don't write default models to db

I'm not 100% sure about any of this but I wanted to get this up for review before too long. Please feel free to tell me to change anything at all! It should be in_roughly_ reviewable commits.

## Types of changes

What types of changes does your code introduce?
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [ ] I have read the [guides](https://github.com/mozilla-it/ctms-api/tree/main/guides)
- [ ] I have followed the [Mozilla Lean Data Policies](https://www.mozilla.org/en-US/about/policy/lean-data/)
- [ ] Lint and tests pass both locally and on the cicd with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: tests/unit/test_api.py [hunk 1]
```diff
@@ -334,117 +335,109 @@ def test_get_ctms_by_alt_id_none_found(client, dbsession, alt_id_name, alt_id_va
...
+    )
+    assert saved_contacts[0].email.mailing_country == sample.email.mailing_country
 
 
-def test_create_basic_with_basket_collision(client, dbsession):
+@pytest.mark.parametrize("add_contact", SAMPLE_CONTACTS.keys(), indirect=True)
+def test_create_basic_with_basket_collision(add_contact):
     """Creating a contact with diff ids but same email fails.
     We override the basket token so that we know we're not colliding on that here.
     See other test for that check
```
Comment (jwhitlock) [inline tests/unit/test_api.py:410]: Old code, but could this comment be
```suggestion
    See other test_create_basic_with_email_collision below for that check
```

**Review by jwhitlock — APPROVED** [review]
Thanks @imbstack!

When I POST with a minimal body ``{"email": {"primary_email": "email@example.com"}}``, I get an error that ``basket_token`` and ``email_id`` are required. I think the field definitions need a ``default=None`` to work.

I think all these suggestions are nits and could be done in future PRs.

File: ctms/crud.py [hunk 2]
```diff
@@ -20,6 +24,22 @@ def get_email_by_email_id(db: Session, email_id: UUID4):
     return db.query(Email).filter(Email.email_id == email_id).first()
 
 
+def get_amo_by_email_id(db: Session, email_id: UUID4):
+    return db.query(AmoAccount).filter(AmoAccount.email_id == email_id).first()
```
[the remark this answers is not in the document]
Comment (imbstack) [reply ctms/crud.py:28]: I'll get this in a followup!

File: ctms/app.py [hunk 3]
```diff
@@ -185,6 +185,7 @@ def read_ctms_by_email_id(
 @app.post(
     "/ctms",
     summary="Create a contact, generating an id",
+    responses={409: {"model": BadRequestResponse}},
```
[the remark this answers is not in the document]
Comment (imbstack) [reply ctms/app.py:188]: Also will get this in a followup!

**Pull request merged**